File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

float get_positive_float(void);

int main(void)
{
    float dollar = get_positive_float();
    // convert dollar to cents to avoid floating point precision error
    int cent = round(dollar * 100);
    // count variable to keep count of coins
    int count = 0;
    // Loop for getting count of 25 cent coins
    while (cent >= 25)
    {
        count++;
        cent = cent - 25;
    }
    // Loop for getting count of 10 cent coins
    while (cent >= 10 && cent < 25)
    {
        count++;
        cent = cent - 10;
    }
    // Loop for getting count of 5 cent coins
    while (cent >= 5 && cent < 10)
    {
        count++;
        cent = cent - 5;
    }
    // Loop for getting count of 1 cent coins
    while (cent >= 1 && cent < 5)
    {
        count++;
        cent = cent - 1;
    }
    printf("%i", count);
}

float get_positive_float(void)
{
    // Keeping asking for input till user does not enter a positive number.
    float n;
    do
    {
        n = get_float("Change owed:\n");
    }
    while (n < 0.0);
    return n;
}

File: pset6/cash/cash.py
from cs50 import get_float

# Prompt for input while input is not valid
while True:
    cash = get_float("Change owed:")
    if cash > 0:
        break

# Convert dollars to cents
change = round(cash * 100)

# Initialize count
count = 0

# If change is more than 25 cents
if change > 25:
    count += int(change / 25)
    change = change % 25

# If change is less than 25 cents but more than 10 cents
if change >= 10 and change < 25:
    count += int(change / 10)
    change = change % 10

# If change is less than 10 cents but more than 5 cents
if change >= 5 and change < 10:
    count += int(change / 5)
    change = change % 5

# If change is less than 5 cents but more than 1 cent
if change >= 1 and change < 5:
    count += int(change / 1)
    change = change % 1

print(count)